source/Octostache/Templates/Functions/TextManipulationFunction.cs

```diff
@@ -95,5 +96,61 @@ public static string Trim(string argument, string[] options)
                     return null;
             }
         }
+
+        public static string UriPart(string argument, string[] options)
+        {
+            if (argument == null)
+                return null;
+
+            if (!options.Any())
+                return $"[{nameof(UriPart)}: no argument given]";
```

Comment:
I actually like the approach of emitting error messages into the result. Other filters don't do that, instead choosing to silently-fail. This approach may cause some confusion when it emits an error message, but finding the root cause of a problem will prove easier in the end, especially if a log message is written by a later line of script, or the person logs out the evaluated variables as part of the deployment.